### PR TITLE
Simple project now working with typescript

### PR DESCRIPTION
**Pull request by cairesvs**
# Problem

Fix the problems related to serverless with aws lambda using nextjs and typescript.

# Solution

Migrate project from JS to Typescript and fix the serverless config with nextjs.

The idea is to have an example that explores the power of serverless, nextjs. The typescript part is well separated so would be easy to run with javascript (we always run on top of javascript generated on dist directory)

### CODE REVIEW

File: src/functions/routeHandler.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+function createServer(req: any, res: any) {
+	console.log(`req from ${req.headers.host}`)
+	handle(req, res)
+}
+
+if (!isHostedOnAWS) {
+	app.prepare().then(() => {
+		const server = express()
+		server.get('*', (req, res) => handle(req, res))
+		server.listen(3000)
```
Comment (Vadorequest) [inline src/functions/routeHandler.ts:20]: Hard coded port

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			"es2015.symbol",
+			"es2015.symbol.wellknown",
+			"esnext.asynciterable"
+		],
+		"module": "commonjs",
+		"target": "es6",
+		"jsx": "react-native",
+		"strict": true,
+		"moduleResolution": "node",
+		"sourceMap": false,
```
Comment (Vadorequest) [inline tsconfig.json:23]: No source maps with TS? Wanted behavior?

**Comment by Vadorequest** [comment]
@cairesvs Thanks for the contribution, I didn't receive any notification from your PR, hence the reason for late feedback.

I don't want to merge this into master, I'd rather clean master from its current bugs (if possible...), and make a separate branch with TS support, not everybody wants to work with TS and I believe that's the cleanest approach. What do you think?

I just pushed some stuff into master, which created conflicts. Mostly with packages I believe.

Did you try it only in local environment or also on AWS? I saw you used Express, which is great as I failed to use it and make it work so sub routes would be handled by the same root handler. Looks like you've succeeded! 👍

File: src/functions/routeHandler.ts (hunk 1 above)
Comment (cairesvs) [reply src/functions/routeHandler.ts:20]: Thanks! I forgot to use the `port` variable here.

File: tsconfig.json (hunk 2 above)
Comment (cairesvs) [reply tsconfig.json:23]: This was related to a test but this should be true

**Comment by cairesvs** [comment]
@Vadorequest Makes total sense! My main goal here was to provide some kind of material to help you with the problems that I was facing (similar problems that you're facing)

**Pull request closed without merging**